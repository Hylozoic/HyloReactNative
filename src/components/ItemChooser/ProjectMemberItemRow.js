import React from 'react'
import {
  Text,
  TouchableOpacity
} from 'react-native'
import { caribbeanGreen } from 'style/colors'
import { propTypesForItemRowComponent } from '../ItemChooser'
import Avatar from '../Avatar'
import RoundCheckbox from 'rn-round-checkbox'

export default function ProjectMemberItemRow ({
  item: person,
  chosen,
  toggleChosen
}) {
  return (
    <TouchableOpacity style={styles.personRow} onPress={() => toggleChosen(person)}>
      <Avatar style={styles.personAvatar} avatarUrl={person.avatarUrl} dimension={30} />
      <Text style={styles.personName}>{person.name}</Text>
      <RoundCheckbox
        style={styles.checkbox}
        checked={chosen}
        backgroundColor={caribbeanGreen}
        onValueChange={() => toggleChosen(person)}
      />
    </TouchableOpacity>
  )
}
ProjectMemberItemRow.propTypes = propTypesForItemRowComponent

const styles = {
  personRow: {
    padding: 15,
    paddingBottom: 0,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'flex-start'
  },
  personAvatar: {
    marginRight: 12
  },
  personName: {
    fontFamily: 'Circular-Bold',
    flex: 1
  },
  checkbox: {
    marginLeft: 'auto',
    width: 50
  }
}
