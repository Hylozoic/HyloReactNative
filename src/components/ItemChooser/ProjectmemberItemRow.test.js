import React from 'react'
import ReactShallowRenderer from 'react-test-renderer/shallow'
import ProjectMemberItemRow from './ProjectMemberItemRow'

it('renders as expected', () => {
  const props = {
    item: { id: 1 },
    onPress: () => {}
  }
  const renderer = new ReactShallowRenderer()
  renderer.render(<ProjectMemberItemRow {...props} />)
  expect(renderer.getRenderOutput()).toMatchSnapshot()
})
