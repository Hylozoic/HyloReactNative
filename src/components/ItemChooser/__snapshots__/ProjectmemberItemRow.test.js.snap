// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as expected 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[Function]}
>
  <Avatar
    avatarUrl={undefined}
    dimension={30}
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  />
  <RoundCheckbox
    backgroundColor="#0DC39F"
    borderColor="grey"
    checked={false}
    icon="ios-checkmark"
    iconColor="white"
    onValueChange={[Function]}
    size={24}
  />
</TouchableOpacity>
`;
